proposed/http-factory/http-factory.md

```diff
@@ -163,14 +170,17 @@ fwrite($resource, $body);
 Has the ability to create streams for uploaded files.
 
 ```php
-namespace Psr\Http\Message;
+namespace Psr\Http\Factory;
+
+use Psr\Http\Message\UploadedFileInterface;
 
 interface UploadedFileFactoryInterface
 {
     /**
      * Create a new uploaded file.
      *
-     * If a string is passed it is assumed to be a file path.
+     * If a string is used to create the file, a temporary resource will be
+     * created with the content of the string.
```

Comment:
@weierophinney I tried to follow through with this and got really tied up trying to deal variations in the parameters between different methods. Largely the same type of confusion we had in trying to define the stream factory methods. It seems like there should actually be several methods:

- create from file path ($_FILES compatibility)
- create from string of content
- create from resource
- create from stream

The `$error` parameter really bothers me. In any place except `$_FILES` compatibility, does it matter? Why wouldn't the result always be `UPLOAD_ERR_OK`?